cmd/podman/diff.go

```diff
@@ -33,31 +33,35 @@ func (so stdoutStruct) Out() error {
 }
 
 var (
-	diffFlags = []cli.Flag{
-		cli.BoolFlag{
-			Name:   "archive",
-			Usage:  "Save the diff as a tar archive",
-			Hidden: true,
-		},
-		cli.StringFlag{
-			Name:  "format",
-			Usage: "Change the output format.",
-		},
-	}
+	diffCommand     cliconfig.DiffValues
 	diffDescription = fmt.Sprint(`Displays changes on a container or image's filesystem.  The
 	container or image will be compared to its parent layer`)
 
-	diffCommand = cli.Command{
-		Name:         "diff",
-		Usage:        "Inspect changes on container's file systems",
-		Description:  diffDescription,
-		Flags:        sortFlags(diffFlags),
-		Action:       diffCmd,
-		ArgsUsage:    "ID-NAME",
-		OnUsageError: usageErrorHandler,
+	_diffCommand = &cobra.Command{
+		Use:   "diff",
+		Short: "Inspect changes on container's file systems",
+		Long:  diffDescription,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			diffCommand.InputArgs = args
+			diffCommand.GlobalFlags = MainGlobalOpts
+			return diffCmd(&diffCommand)
+		},
+		Example: "ID-NAME",
 	}
 )
 
+func init() {
+	diffCommand.Command = _diffCommand
+	flags := diffCommand.Flags()
+
+	flags.BoolVar(&diffCommand.Archive, "archive", true, "Save the diff as a tar archive")
+	flags.StringVar(&diffCommand.Format, "format", "", "Change the output format.")
```

Comment:
although I prefer periods at the end, I think we decided to leave them off the last line.  Regardless we ought to be consistent.